WooCommerce/Classes/ViewRelated/Orders/OrdersRootViewController.swift

```diff
@@ -228,6 +252,49 @@ private extension OrdersRootViewController {
         }
     }
 
+    func configureUpsellCardReaderFeatureAnnouncement() {
+        guard upsellCardReadersAnnouncementViewModel.shouldBeVisible else {
+            return
+        }
+
+        let view = FeatureAnnouncementCardView(viewModel: upsellCardReadersAnnouncementViewModel,
+                                               dismiss: { [weak self] in
+            self?.hideUpsellCardReaderFeatureAnnouncementView(true)
+        }, callToAction: {
+            let configuration = CardPresentConfigurationLoader().configuration
+            WebviewHelper.launch(configuration.purchaseCardReaderUrl(), with: self)
+        })
+            .background(Color(.listForeground))
+
+        let hostingViewController = ConstraintsUpdatingHostingController(rootView: view)
+
+        guard let hostingView = hostingViewController.view else {
+            return
+        }
+
+        stackView.setCustomSpacing(UIStackView.spacingUseSystem, after: filtersBar)
+
+        addChild(hostingViewController)
+        stackView.addArrangedSubview(hostingView)
+        hostingViewController.didMove(toParent: self)
+
+        upsellCardReaderFeatureAnnouncementViewController = hostingViewController
+    }
+
+    func updateUpsellCardUpsellCardReaderFeatureAnnouncementVisibility() {
+        let shouldBeShown = UIDevice.current.orientation.isPortrait && upsellCardReadersAnnouncementViewModel.shouldBeVisible
```

Comment:
I wonder if we should use a `UITraitCollection` and target the `verticalSizeClass` here rather than checking for `.isPortrait`? This way we avoid possible conflicts with users locking the interface position, or when using split views.

Reply:
Great point 👏 ! Originally I didn't use the `UITraitCollection` because I thought there were some cases with `verticalSizeClass` `.regular` where we wouldn't like to show the banner. After changing it and checking for all the cases I realised it is not the case, so I changed it https://github.com/woocommerce/woocommerce-ios/pull/7326/commits/5fafcddf3df436bce231bc59f5b99ee5fb845f65